MigrationExecutorApp/ChangeFeedProcessortHost.cs

```diff
@@ -108,92 +121,131 @@ public async Task CloseAsync()
-                processorOptions.StartFromBeginning,
-                processorOptions.LeaseRenewInterval.ToString());
+            await changeFeedProcessor.StartAsync().ConfigureAwait(false);
 
-            processorOptions.MaxItemCount = 1000;
-            var destClient = new DocumentClient(destCollInfo.Uri, destCollInfo.MasterKey,
-                new ConnectionPolicy() { ConnectionMode = ConnectionMode.Direct, ConnectionProtocol = Protocol.Tcp },
-                ConsistencyLevel.Eventual);
+            return changeFeedProcessor;
+        }
 
-            var docTransformer = new DefaultDocumentTransformer();
+        async Task ProcessChangesAsync(IReadOnlyCollection<Document> docs, CancellationToken cancellationToken)
+        {
+            Boolean isSyntheticKey = SourcePartitionKeys.Contains(",");
+            Boolean isNestedAttribute = SourcePartitionKeys.Contains("/");
+            Container targetContainer = destinationCollectionClient.GetContainer(config.DestDbName, config.DestCollectionName);
+            containerToStoreDocuments = targetContainer;
+            ConcurrentDictionary<int, int> failedMetrics = new ConcurrentDictionary<int, int>();
+            List<Task> tasks = new List<Task>();
+            List<Document> failedDocs = new List<Document>();
+            Stopwatch stopwatch = Stopwatch.StartNew();
+            Document document = new Document();
+            foreach (Document doc in docs)
+            {
+                Console.WriteLine($"\tDetected operation...");
+                document = (SourcePartitionKeys != null & TargetPartitionKey != null) ? MapPartitionKey(doc, isSyntheticKey, TargetPartitionKey, isNestedAttribute, SourcePartitionKeys) : document = doc;
+                tasks.Add(containerToStoreDocuments.CreateItemAsync(item: document, cancellationToken: cancellationToken).ContinueWith((Task<ItemResponse<Document>> task) =>
+                {
+                    AggregateException innerExceptions = task.Exception.Flatten();
+                    CosmosException cosmosException = (CosmosException)innerExceptions.InnerExceptions.FirstOrDefault(innerEx => innerEx is CosmosException);
+                    failedMetrics.AddOrUpdate((int)cosmosException.StatusCode, 0, (key, value) => value + 1);
+                    WriteFailedDocsToBlob("FailedDocImport", containerClient, cosmosException, doc);
+                }, TaskContinuationOptions.OnlyOnFaulted));
+                // Simulate work
+                await Task.Delay(1);
+            }
+        }
 
-            BlobContainerClient containerClient = null;
+        private static void WriteFailedDocsToBlob(string failureType, BlobContainerClient containerClient, CosmosException cosmosException, Document doc)
+        {
+            string failedDoc;
+            byte[] byteArray;
+            BlobClient blobClient = containerClient.GetBlobClient(failureType + Guid.NewGuid().ToString() + ".csv");
```

Comment:
This creates a blobclient for each failure, are these BlobClient things that need to be disposed or can be cached?

Reply:
@revinjchalil - can you comment on this?

Reply:
The BlobClient uses an instance of https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient?view=netcore-3.1 to send HTTP requests to the Blob service.  The HTTP client that's being used is the shared instance and all instances of BlobClient use the same instance of HTTP client and so there are no additional resources being allocated that needs disposal.